frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/helper-text-strings.ts

```diff
@@ -1,39 +1,55 @@
 import { t } from "ttag";
-import { HelpText } from "metabase-lib/expressions/types";
+import moment from "moment-timezone";
 
-const helperTextStrings: HelpText[] = [
+import { formatTime } from "metabase/lib/formatting";
+import type { Database } from "metabase-types/types/Database";
+import { HelpText, HelpTextConfig } from "metabase-lib/expressions/types";
+
+const getDescriptionForNow = (database: Database, reportTimezone: string) => {
+  const hasTimezoneFeatureFlag = database.features.includes("set-timezone");
+  const timezone = hasTimezoneFeatureFlag ? reportTimezone : "UTC";
+  const nowAtTimezone = getNowAtTimezone(timezone, reportTimezone);
+
+  if (database.engine === "h2") {
```

Comment:
Why do we need this special case here? Shouldn't it work based on features that the database support only?

Reply:
@qnkhuat could you take a look at this? It seems that this hack exists because of some mismatch between FE/BE.

Reply:
@ranquild, 

@calherries has been on this.
He'd left a comment here explaining how h2 works differently.
Github seems to have changed how it displays comments after the lines commented have been updated.
And I myself can't find the comment here anymore.

@calherries, can you refresh us on the reasoning for us showing the shorter description for `h2`, please?

Reply:
The special case is here because H2 is the only DBMS we support where:
- set-timezone isn't a feature, and 
- It's possible for `now` to be in the system timezone, not UTC.

Its possible to have this as a feature, but I didn't think it was worth it. We should definitely add this as a comment though.